src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.cs

```diff
@@ -43,17 +43,33 @@ public void MethodWithArrayArgument() { }
-            yield return new object[] { "build_property.TargetFramework = netcoreapp1.0", false };
-            yield return new object[] { "build_property.TargetFramework = dotnet", false };
-            yield return new object[] { "build_property.TargetFramework = uap10.0", false };
-            yield return new object[] { "build_property.TargetFramework = netstandard2.1", false };
-            yield return new object[] { "build_property.TargetFramework = net5", true };
-            yield return new object[] { "build_property.TargetFramework = net5.0", true };
-            yield return new object[] { "build_property.TargetFramework = net5.0-windows", true };
-            yield return new object[] { "build_property.TargetFramework = net5.0-ios14.0", true };
-            yield return new object[] { "build_property.TargetFramework = Net99", true };
-            yield return new object[] { "build_property.TargetFramework = netcoreapp5", false };
+            // Pre-.NET 5 TFMS
+            yield return new object[] { "build_property.TargetFramework = net472\nbuild_property.TargetFrameworkIdentifier=.NETFramework\nbuild_property.TargetFrameworkVersion=v4.7.2", false };
+            yield return new object[] { "build_property.TargetFramework = netcoreapp1.0\nbuild_property.TargetFrameworkIdentifier=.NETCoreApp\nbuild_property.TargetFrameworkVersion=v1.0", false };
+            yield return new object[] { "build_property.TargetFramework = dotnet\nbuild_property.TargetFrameworkIdentifier=.NETPlatform\nbuild_property.TargetFrameworkVersion=v5.0", false };
+            yield return new object[] { "build_property.TargetFramework = uap10.0\nbuild_property.TargetFrameworkIdentifier=UAP\nbuild_property.TargetFrameworkVersion=v10.0", false };
+            yield return new object[] { "build_property.TargetFramework = netstandard2.1\nbuild_property.TargetFrameworkIdentifier=.NETStandard\nbuild_property.TargetFrameworkVersion=v2.1", false };
+
+            // .NET 5+ TFMs with a single major version digit
+            yield return new object[] { "build_property.TargetFramework = net5\nbuild_property.TargetFrameworkIdentifier=.NETCoreApp\nbuild_property.TargetFrameworkVersion=v5.0", true };
+            yield return new object[] { "build_property.TargetFramework = net5.0\nbuild_property.TargetFrameworkIdentifier=.NETCoreApp\nbuild_property.TargetFrameworkVersion=v5.0", true };
+            yield return new object[] { "build_property.TargetFramework = net5.0-windows\nbuild_property.TargetFrameworkIdentifier=.NETCoreApp\nbuild_property.TargetFrameworkVersion=v5.0", true };
+            yield return new object[] { "build_property.TargetFramework = net5.0-ios14.0\nbuild_property.TargetFrameworkIdentifier=.NETCoreApp\nbuild_property.TargetFrameworkVersion=v5.0", true };
+            yield return new object[] { "build_property.TargetFramework = netcoreapp5\nbuild_property.TargetFrameworkIdentifier=.NETCoreApp\nbuild_property.TargetFrameworkVersion=v5.0", true };
```

Comment:
A case to note is that `TargetFramework=netcoreapp5` was previously treated as "unsupported" by the analyzer.

However, the actual MSBuild path done by our props/targets recognizes this TFM and infers `TargetFrameworkIdentifier=.NETCoreApp` and `TargetFrameworkVersion=v5.0` hence is equivalent to `TargetFramework=net5.0` and thus is handled by the analyzer under the new code